### ISSUE TITLE
Add quiet option to Node API

### ISSUE THREAD

**Issue by duanehutchins** [stylelint/stylelint#4476]
> Clearly describe the bug

The Node API ignores the `quiet` option.  You can confirm that the `quiet` option is not used in `lib/standalone.js`.

Example gulpfile.js:

```gulpfile.js
gulp.task('stylelint', () => {
    const stylelint = require('stylelint');
    const options = Object.assign(argv, {
        files: 'sass/**/*.scss',
        formatter: 'string',
        quiet: true,
      });
    return stylelint.lint(options).then(data => {
        console.log(data.output);
    });
});
```

Even though `quiet: true`, the warnings are not suppressed.

> Which version of stylelint are you using?

`12.0.0`

> How are you running stylelint: CLI, PostCSS plugin, Node.js API?

Node.js API

> Does the bug relate to non-standard syntax (e.g. SCSS, Less etc.)?

No

> What did you expect to happen?

Warnings should be suppressed in quiet mode.  Example:

```shell
$ npm run stylelint
/src/sass/_mixins.scss:101:2: Expected single space before "{" (block-opening-brace-space-before) [error]

1 problem
```

> What actually happened (e.g. what warnings or errors did you get)?

Warnings are not suppressed.  Example:
```shell
$ npm run stylelint
/src/sass/_mixins.scss:101:2: Expected single space before "{" (block-opening-brace-space-before) [error]
/src/sass/_mixins.scss:234:18: Unexpected !important (declaration-no-important) [warning]
/src/sass/_mixins.scss:238:20: Unexpected !important (declaration-no-important) [warning]

3 problems
```

**Comment by evilebottnawi** [comment]
i think `quiet` option only for CLI

**Comment by jeddy3** [comment]
Is there any reason we don't allow it to be used by the Node API? If not, we could add it.

**Comment by jeddy3** [comment]
As it's a valid use case (where a user wants `data.output` to be filtered of errors when using the Node.js API) and there are no objections, I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time. 

You'll need to move `quiet` out from [the CLI docs](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/usage/cli.md) and into the [Options](https://github.com/stylelint/stylelint/blob/master/docs/user-guide/usage/options.md) docs in the pull request.

**Comment by glen-84** [comment]
@jeddy3 I think this should be closed? (#5542)
